tests/fs/api.private.test.ts

```diff
@@ -0,0 +1,284 @@
+          const filepath = path.file('private', data[0])
+          const [val] = data[1]
+
+          await fs.write(filepath, val)
+          await fs.rm(filepath)
+
+          expect(await fs.exists(filepath)).toEqual(false)
+        })
+    )
+  })
+
+  it('reads files it writes', async () => {
+    const fs = await emptyFilesystem()
+
+    await fc.assert(
+      fc.asyncProperty(
+        fc.tuple(pathSegment(), fileContent()), async data => {
+          const filepath = path.file('private', data[0])
+          const [val, type] = data[1]
+
+          await fs.write(filepath, val)
+          const file = await fs.read(filepath)
+          const content = maybeDecode(file, type)
+
+          expect(content).toEqual(val)
+        })
+    )
+  })
+
+  it('moves files', async () => {
+    const fs = await emptyFilesystem()
+
+    await fc.assert(
+      fc.asyncProperty(
+        fc.tuple(pathSegmentPair(), fileContent()), async data => {
+          const [from, to] = data[0]
+          const [val] = data[1]
+          const fromPath = path.file('private', from)
+          const toPath = path.file('private', to)
+
+          await fs.write(fromPath, val)
+          await fs.mv(fromPath, toPath)
+          const fromExists = await fs.exists(fromPath)
+          const toExists = await fs.exists(toPath)
+
+          expect(toExists && !fromExists).toEqual(true)
+        })
+    )
+  })
+
+  it('reads moved files', async () => {
+    const fs = await emptyFilesystem()
+
+    await fc.assert(
+      fc.asyncProperty(
+        fc.tuple(pathSegmentPair(), fileContent()), async data => {
+          const [from, to] = data[0]
+          const [val, type] = data[1]
+          const fromPath = path.file('private', from)
+          const toPath = path.file('private', to)
+
+          await fs.write(fromPath, val)
+          await fs.mv(fromPath, toPath)
+
+          const file = await fs.read(toPath)
+          const content = maybeDecode(file, type)
+
+          expect(content).toEqual(val)
+        })
+    )
+  })
+
+  it('makes directories', async () => {
+    const fs = await emptyFilesystem()
+
+    await fc.assert(
+      fc.asyncProperty(
+        pathSegment(), async data => {
+          const dirpath = path.directory('private', data[0])
+
+          await fs.mkdir(dirpath)
+
+          expect(await fs.exists(dirpath)).toEqual(true)
+        })
+    )
+  })
+
+  it('removes directories it makes', async () => {
+    const fs = await emptyFilesystem()
+
+    await fc.assert(
+      fc.asyncProperty(
+        pathSegment(), async data => {
+          const dirpath = path.directory('private', data[0])
+
+          await fs.mkdir(dirpath)
+          await fs.rm(dirpath)
+
+          expect(await fs.exists(dirpath)).toEqual(false)
+        })
+    )
+  })
+
+  it('writes files to a directory', async () => {
+    const fs = await emptyFilesystem()
+    const dirpath = path.directory('private', 'testDir')
+    await fs.mkdir(dirpath)
```

Comment:
You could even make the `mkdir` optional here, so that it also tests if `fs.write` creates the parent dir.

Reply:
Good call. Removing the `mkdir` from this totally makes sense.